mmcls/models/backbones/mlp_mixer.py

```diff
@@ -0,0 +1,218 @@
+                'num_layers': 24,
+                'tokens_mlp_dims': 512,
+                'channels_mlp_dims': 4096,
+            }),
+    }
+
+    def __init__(self,
+                 arch='b',
+                 img_size=224,
+                 patch_size=16,
+                 drop_rate=0.,
+                 drop_path_rate=0.,
+                 norm_cfg=dict(type='LN'),
+                 act_cfg=dict(type='GELU'),
+                 patch_cfg=dict(),
+                 layer_cfgs=dict(),
+                 init_cfg=None):
+        super(MlpMixer, self).__init__(init_cfg)
+
+        if isinstance(arch, str):
+            arch = arch.lower()
+            assert arch in set(self.arch_zoo), \
+                f'Arch {arch} is not in default archs {set(self.arch_zoo)}'
+            self.arch_settings = self.arch_zoo[arch]
+        else:
+            essential_keys = {
+                'embed_dims', 'num_layers', 'tokens_mlp_dims',
+                'channels_mlp_dims'
+            }
+            assert isinstance(arch, dict) and set(arch) == essential_keys, \
+                f'Custom arch needs a dict with keys {essential_keys}'
+            self.arch_settings = arch
+
+        self.embed_dims = self.arch_settings['embed_dims']
+        self.num_layers = self.arch_settings['num_layers']
+        self.tokens_mlp_dims = self.arch_settings['tokens_mlp_dims']
+        self.channels_mlp_dims = self.arch_settings['channels_mlp_dims']
+
+        self.img_size = to_2tuple(img_size)
+
+        _patch_cfg = dict(
+            img_size=img_size,
+            embed_dims=self.embed_dims,
+            conv_cfg=dict(
+                type='Conv2d', kernel_size=patch_size, stride=patch_size),
+        )
+        _patch_cfg.update(patch_cfg)
+        self.patch_embed = PatchEmbed(**_patch_cfg)
+        num_patches = self.patch_embed.num_patches
+
+        self.layers = ModuleList()
+        if isinstance(layer_cfgs, dict):
+            layer_cfgs = [layer_cfgs] * self.num_layers
+        for i in range(self.num_layers):
+            _layer_cfg = dict(
+                num_tokens=num_patches,
+                embed_dims=self.embed_dims,
+                tokens_mlp_dims=self.tokens_mlp_dims,
+                channels_mlp_dims=self.channels_mlp_dims,
+                drop_rate=drop_rate,
+                drop_path_rate=drop_path_rate,
+                act_cfg=act_cfg,
+                norm_cfg=norm_cfg,
+            )
+            _layer_cfg.update(layer_cfgs[i])
+            self.layers.append(MixerBlock(**_layer_cfg))
+
+        self.norm1_name, norm1 = build_norm_layer(
+            norm_cfg, self.embed_dims, postfix=1)
+        self.add_module(self.norm1_name, norm1)
+
+    @property
+    def norm1(self):
+        return getattr(self, self.norm1_name)
+
+    def forward(self, x):
+        x = self.patch_embed(x)
+
+        for layer in self.layers:
+            x = layer(x)
+
+        x = self.norm1(x).transpose(1, 2)
+        return x
```

Comment:
The output should be a tuple instead of a single tensor. Consider using `return (x, )` instead. 

Could you please further support `out_indices` to output the selected indices for more find grained output control?

Reply:
Done. Added `out_indices` to support output control.